File: app/src/main/java/calendario/kevshupp/diariokevinali/Message.java
package calendario.kevshupp.diariokevinali;

import java.util.ArrayList;
import java.util.List;

public class Message {
    public static final String TYPE_MESSAGE = "MESSAGE";
    public static final String TYPE_ALBUM = "ALBUM";

    private String messageId;
    private String partnerId;
    private String authorId;
    private String authorName;
    private String authorImageUrl;
    private String content;
    private String title;
    private List<String> imageUrls;
    private long timestamp;
    private boolean liked;
    private String type = TYPE_MESSAGE;

    public Message() {
        this.imageUrls = new ArrayList<>();
        this.liked = false;
    }

    public Message(String messageId, String partnerId, String authorId, String authorName, String authorImageUrl, String content, List<String> imageUrls, long timestamp, boolean liked) {
        this.messageId = messageId;
        this.partnerId = partnerId;
        this.authorId = authorId;
        this.authorName = authorName;
        this.authorImageUrl = authorImageUrl;
        this.content = content;
        this.imageUrls = imageUrls;
        this.timestamp = timestamp;
        this.liked = liked;
        this.type = (imageUrls != null && imageUrls.size() > 1) ? TYPE_ALBUM : TYPE_MESSAGE;
    }

    public String getTitle() { return title; }
    public void setTitle(String title) { this.title = title; }

    public String getMessageId() { return messageId; }
    public void setMessageId(String messageId) { this.messageId = messageId; }

    public String getPartnerId() { return partnerId; }
    public void setPartnerId(String partnerId) { this.partnerId = partnerId; }

    public String getAuthorId() { return authorId; }
    public void setAuthorId(String authorId) { this.authorId = authorId; }

    public String getAuthorName() { return authorName; }
    public void setAuthorName(String authorName) { this.authorName = authorName; }

    public String getAuthorImageUrl() { return authorImageUrl; }
    public void setAuthorImageUrl(String authorImageUrl) { this.authorImageUrl = authorImageUrl; }

    public String getContent() { return content; }
    public void setContent(String content) { this.content = content; }

    public List<String> getImageUrls() { return imageUrls; }
    public void setImageUrls(List<String> imageUrls) { this.imageUrls = imageUrls; }

    public String getImageUrl() {
        return (imageUrls != null && !imageUrls.isEmpty()) ? imageUrls.get(0) : null;
    }

    public void setImageUrl(String imageUrl) {
        if (this.imageUrls == null) this.imageUrls = new ArrayList<>();
        else this.imageUrls.clear();
        if (imageUrl != null) this.imageUrls.add(imageUrl);
    }

    public long getTimestamp() { return timestamp; }
    public void setTimestamp(long timestamp) { this.timestamp = timestamp; }

    public boolean isLiked() { return liked; }
    public void setLiked(boolean liked) { this.liked = liked; }

    public String getType() { return type; }
    public void setType(String type) { this.type = type; }
}

File: app/src/main/java/calendario/kevshupp/diariokevinali/Message.kt
package calendario.kevshupp.diariokevinali

import com.google.firebase.firestore.Exclude
import com.google.firebase.firestore.IgnoreExtraProperties
import com.google.firebase.firestore.PropertyName
import androidx.compose.runtime.Stable

/**
 * Modelo de mensaje/publicación para el feed y álbumes.
 */
@IgnoreExtraProperties
@Stable
data class Message @JvmOverloads constructor(
    var messageId: String? = null,
    var partnerId: String? = null,
    var authorId: String? = null,
    var authorName: String? = null,
    var authorImageUrl: String? = null,
    var content: String? = null,
    var title: String? = null,
    var imageUrls: MutableList<String>? = mutableListOf(),
    var timestamp: Long = 0,
    var liked: Boolean = false,
    var type: String? = TYPE_MESSAGE
) {
    constructor(
        messageId: String?,
        partnerId: String?,
        authorId: String?,
        authorName: String?,
        authorImageUrl: String?,
        content: String?,
        imageUrls: MutableList<String>?,
        timestamp: Long,
        liked: Boolean
    ) : this(
        messageId, partnerId, authorId, authorName, authorImageUrl,
        content, null, imageUrls ?: mutableListOf(), timestamp, liked,
        if ((imageUrls?.size ?: 0) > 1) TYPE_ALBUM else TYPE_MESSAGE
    )

    companion object {
        const val TYPE_MESSAGE = "MESSAGE"
        const val TYPE_ALBUM = "ALBUM"
    }

    // Propiedad calculada para compatibilidad con código existente que espera un solo String
    @get:PropertyName("imageUrl")
    @set:PropertyName("imageUrl")
    var imageUrl: String?
        get() = imageUrls?.firstOrNull()
        set(value) {
            if (imageUrls == null) imageUrls = mutableListOf()
            if (value != null && imageUrls?.contains(value) != true) {
                if (imageUrls!!.isEmpty()) {
                    imageUrls?.add(value)
                }
            }
        }

    @get:JvmName("getIsLikedProp")
    @set:JvmName("setIsLikedProp")
    @get:PropertyName("isLiked")
    @set:PropertyName("isLiked")
    var isLiked: Boolean
        get() = liked
        set(value) {
            liked = value
        }

    // Inicialización para ajustar el tipo basado en las imágenes (copiado de la lógica Java)
    init {
        if ((imageUrls?.size ?: 0) > 1) {
            type = TYPE_ALBUM
        }
    }
}
